### PR TITLE
Versioning in jars.

### PR DESCRIPTION
**Pull request by TheSenPie**
Shows git commit hash and branch in title. 
todo, should use some actual versioning logic.

### CODE REVIEW

File: editor-desktop/build.gradle [hunk 1]
```diff
@@ -67,3 +85,148 @@ task distLegacyTest(type: Jar) {
...
+
+        addStringField(classSpecBuilder, "version", version)
+        addIntField(classSpecBuilder, "versionCode", versionCode)
+//        addBooleanField(classSpecBuilder, "debugMode", !releaseMode)
+//        addBooleanField(classSpecBuilder, "resetUserOnLogin", resetUserOnLogin)
+//        addStringsField(classSpecBuilder, "serverUrls", serverUrlArray)
+//        addStringField(classSpecBuilder, "guildChatServiceUrl", guildChatMicroserviceURL)
+//        addBooleanField(classSpecBuilder, "isBetaVersion", isBetaVersion)
+//        addBooleanField(classSpecBuilder, "isPrServer", isPrServer)
+//        addStringField(classSpecBuilder, "SERVER_CONTEXT", serverContext)
```
Comment (vuvkar) [inline editor-desktop/build.gradle:196]: delete

File: build.gradle [hunk 2]
```diff
@@ -43,3 +43,71 @@ allprojects {
...
+        this.commit = grgit.head().abbreviatedId
+        this.branch = grgit.branch.current().getName()
+        this.date = new Date().toLocaleString()
+    }
+
+    String getIOSVersion () {
+        def string = this.versionString
+        String[] split = string.split("\\.")
+        return "${Integer.parseInt(split[0]) + 2}.${split[1]}.${split[2]}"
+    }
```
Comment (vuvkar) [inline build.gradle:103]: We will never have Talos in iOS :D

File: editor-desktop/build.gradle [hunk 3]
```diff
@@ -67,3 +85,148 @@ task distLegacyTest(type: Jar) {
...
+                        "      \\:\\  \\     /:/\\:\\  \\     /:/  /    /:/\\:\\  \\     /:/\\ \\  \\  \n" +
+                        "      /::\\  \\   /::\\~\\:\\  \\   /:/  /    /:/  \\:\\  \\   _\\:\\~\\ \\  \\ \n" +
+                        "     /:/\\:\\__\\ /:/\\:\\ \\:\\__\\ /:/__/    /:/__/ \\:\\__\\ /\\ \\:\\ \\ \\__\n" +
+                        "    /:/  \\/__/ \\/__\\:\\/:/  / \\:\\  \\    \\:\\  \\ /:/  / \\:\\ \\:\\ \\/__/\n" +
+                        "   /:/  /           \\::/  /   \\:\\  \\    \\:\\  /:/  /   \\:\\ \\:\\__\\  \n" +
+                        "   \\/__/            /:/  /     \\:\\  \\    \\:\\/:/  /     \\:\\/:/  /  \n" +
+                        "                   /:/  /       \\:\\__\\    \\::/  /       \\::/  /   \n" +
+                        "                   \\/__/         \\/__/     \\/__/         \\/__/    \n")
+                .build()
+
```
Comment (vuvkar) [inline editor-desktop/build.gradle:145]: This is fancy right? 👯

File: version.properties [hunk 4]
```diff
@@ -0,0 +1,3 @@
+#Wed Oct 27 12:41:13 UTC 2021
+VERSION_BUILD_CODE=0
+VERSION_BUILD_STRING=0.00.0
```
Comment (vuvkar) [inline version.properties:3]: Why do you need this file?

File: editor-desktop/src/com/talosvfx/talos/TalosLauncher.java [hunk 5]
```diff
@@ -58,14 +58,18 @@ public static void main (String[] arg) {
 		config.setBackBufferConfig(1,1,1,1,8,8, 0);
 		config.setWindowIcon("icon/talos-64x64.png");
 
+		// TODO: 16.01.23 use actual build version
+//		String misc = "Version: [" + TALOS_BUILD.getVersion() + "] Branch: [" + TALOS_BUILD.getBranch() + "] Commit:[" + TALOS_BUILD.getCommit() + "]";
```
Comment (vuvkar) [inline editor-desktop/src/com/talosvfx/talos/TalosLauncher.java:62]: Delete comment

**Review by vuvkar — CHANGES_REQUESTED** [review]

File: editor-desktop/build.gradle (hunk 3 above)
Comment (TheSenPie) [reply editor-desktop/build.gradle:145]: yes, fancy header for fancy editor 😏

File: version.properties (hunk 4 above)
Comment (Tom-Ski) [reply version.properties:3]: This should not be used. Its only used for applications that need them.  In our case, we already have proper versioning that is coming from gradle.properties. THis is what controls the versioning for the artifacts when they are being deployed

File: editor-desktop/src/com/talosvfx/talos/TALOS_BUILD.java [hunk 6]
```diff
@@ -0,0 +1,67 @@
+package com.talosvfx.talos;
```
Comment (Tom-Ski) [inline editor-desktop/src/com/talosvfx/talos/TALOS_BUILD.java:1]: Should be in editor project, not editor-desktop

File: editor-desktop/build.gradle [hunk 7]
```diff
@@ -67,3 +85,148 @@ task distLegacyTest(type: Jar) {
...
+task generateBuildConfig {
+
+    description 'Generate build config java file'
+
+    outputs.upToDateWhen { false }
+
+    doFirst {
+        String version = project.versionString
+        int versionCode = project.versionCode
```
Comment (Tom-Ski) [inline editor-desktop/build.gradle:128]: Don't need version code. version should come from project version in gradle.properties.

File: editor-desktop/build.gradle [hunk 8]
```diff
@@ -8,7 +23,10 @@ dependencies {
     implementation libs.logback.classic
 
     implementation variantOf(libs.tiny.gizmo) { classifier("natives-desktop") }
+    implementation libs.javapoet
 
+    compileOnly 'org.projectlombok:lombok:1.18.22'
```
Comment (Tom-Ski) [inline editor-desktop/build.gradle:28]: Don't hard code dependenices like these. It should be done using libs as every other dep is done. idk why you even have lombok here

File: build.gradle (hunk 2 above)
Comment (Tom-Ski) [reply build.gradle:103]: Aha, delete

**Review by Tom-Ski — CHANGES_REQUESTED** [review]

File: build.gradle (hunk 2 above)
Comment (TheSenPie) [reply build.gradle]: done

File: version.properties (hunk 4 above)
Comment (TheSenPie) [reply version.properties]: @Tom-Ski could you check again please. I have changed it to use talos-version.properties, shall I change it to gradle.properties?

File: editor-desktop/src/com/talosvfx/talos/TalosLauncher.java (hunk 5 above)
Comment (TheSenPie) [reply editor-desktop/src/com/talosvfx/talos/TalosLauncher.java]: done

File: editor-desktop/src/com/talosvfx/talos/TALOS_BUILD.java (hunk 6 above)
Comment (TheSenPie) [reply editor-desktop/src/com/talosvfx/talos/TALOS_BUILD.java:1]: oh okay, lemme fix

Comment (TheSenPie) [reply editor-desktop/src/com/talosvfx/talos/TALOS_BUILD.java:1]: shall I also move the task "generateBuildConfig" from :editor-desktop to :editor @Tom-Ski

Comment (Tom-Ski) [reply editor-desktop/src/com/talosvfx/talos/TALOS_BUILD.java:1]: yes

Comment (TheSenPie) [reply editor-desktop/src/com/talosvfx/talos/TALOS_BUILD.java]: fixed

File: editor-desktop/build.gradle (hunk 8 above)
Comment (TheSenPie) [reply editor-desktop/build.gradle]: fixed

**Review by Tom-Ski — APPROVED** [review]

**Pull request merged**